llvm/include/llvm/MC/MCPseudoProbe.h

```diff
@@ -276,17 +278,35 @@ class MCPseudoProbeInlineTree
 };
 
 // inline tree node for the decoded pseudo probe
-class MCDecodedPseudoProbeInlineTree
-    : public MCPseudoProbeInlineTreeBase<MCDecodedPseudoProbe *,
-                                         MCDecodedPseudoProbeInlineTree> {
+class MCDecodedPseudoProbeInlineTree {
```

Comment:
Now that this class is no longer derived from `MCPseudoProbeInlineTreeBase`,  `MCPseudoProbeInlineTree ` will be the only derived class of`MCPseudoProbeInlineTreeBase`, there is no need for the base class? we may need to refactor it.